### PR TITLE
Single node v2

### PR DESCRIPTION
**Pull request by eranco74**
/hold

### CODE REVIEW

File: patch.sh [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    # scale down ingress
+    oc --kubeconfig ./mydir/auth/kubeconfig scale --replicas=1 deployments/router-default -n openshift-ingress || return 1
+
+    oc --kubeconfig ./mydir/auth/kubeconfig patch clusterversion/version --type='merge' -p "$(cat <<- EOF
+ spec:
+    overrides:
+      - group: apps/v1
+        kind: Deployment
+        name: etcd-quorum-guard
+        namespace: openshift-machine-config-operator
```
Comment (eranco74) [inline patch.sh:31]:
```suggestion
        namespace: openshift-etcd
```

File: data/data/bootstrap/files/usr/local/bin/patch.sh [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    oc patch --kubeconfig ./auth/kubeconfig -n openshift-ingress-operator ingresscontroller/default --patch '{"spec":{"replicas": 1}}' --type=merge || return 1
+
+    # Mark etcd-quorum-guard as unmanaged
+    oc --kubeconfig ./auth/kubeconfig patch clusterversion/version --type='merge' -p "$(cat <<- EOF
+ spec:
+    overrides:
+      - group: apps/v1
+        kind: Deployment
+        name: etcd-quorum-guard
+        namespace: openshift-machine-config-operator
```
Comment (eranco74) [inline data/data/bootstrap/files/usr/local/bin/patch.sh:20]: @tsorya @osherdp this should be openshift-etcd

**Comment by gbraad** [comment]
Hi, can you explain what this PR tries to solve and what this aligns with? There is an effort to create a Single Node Cluster at https://github.com/code-ready/snc/ which is used by CRC. Also, for resource tuning, effort is being made with the Single Node Profile enhancement, and is work on by the same team. Would appreciate some back story or more info

File: pkg/asset/tls/tls.go [hunk 3]
```diff
@@ -24,6 +24,9 @@ const (
 	// ValidityOneDay sets the validity of a cert to 24 hours.
 	ValidityOneDay = time.Hour * 24
 
+	// ValidityOneMinute sets the validity of a cert to 1 month.
+	ValidityOneMonth = ValidityOneDay * 30
```
Comment (gbraad) [inline pkg/asset/tls/tls.go:28]: :-/

This was a no-go after discussions. Initial certificate creation was only allowed to be max 24hrs, after which when rotated is extended to the expected 30 days.

File: pkg/asset/tls/kubelet.go [hunk 4]
```diff
@@ -24,7 +24,7 @@ func (c *KubeletCSRSignerCertKey) Generate(parents asset.Parents) error {
 	cfg := &CertCfg{
 		Subject:   pkix.Name{CommonName: "kubelet-signer", OrganizationalUnit: []string{"openshift"}},
 		KeyUsages: x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature | x509.KeyUsageCertSign,
-		Validity:  ValidityOneDay,
+		Validity:  ValidityOneMonth,
```
Comment (gbraad) [inline pkg/asset/tls/kubelet.go:27]: Ditto; https://github.com/openshift/installer/pull/4308/files#r530360087

File: data/data/bootstrap/files/usr/local/bin/patch.sh [hunk 5]
```diff
@@ -0,0 +1,36 @@
+#!/bin/bash -x
```
Comment (marun) [inline data/data/bootstrap/files/usr/local/bin/patch.sh:1]: What is the relationship between this file and patch.sh?

Comment (eranco74) [reply data/data/bootstrap/files/usr/local/bin/patch.sh:1]: The patch.sh is the initial version I had when I started running this manually.
forgut to remove it, thx

**Comment by eranco74** [comment]
> Hi, can you explain what this PR tries to solve and what this aligns with? There is an effort to create a Single Node Cluster at https://github.com/code-ready/snc/ which is used by CRC. Also, for resource tuning, effort is being made with the Single Node Profile enhancement, and is work on by the same team. Would appreciate some back story or more info

This PR is in the context of https://github.com/openshift/enhancements/pull/504
As part of VZ far edge initiative, we’re expected to deliver Single Node Openshift POC to VZ lab by Dec 13, 2020.
This is the installer branch we use for the installation.

**Comment by cfergeau** [comment]
> > Hi, can you explain what this PR tries to solve and what this aligns with? There is an effort to create a Single Node Cluster at https://github.com/code-ready/snc/ which is used by CRC. Also, for resource tuning, effort is being made with the Single Node Profile enhancement, and is work on by the same team. Would appreciate some back story or more info
> 
> This PR is in the context of [openshift/enhancements#504](https://github.com/openshift/enhancements/pull/504)
> As part of VZ far edge initiative, we’re expected to deliver Single Node Openshift POC to VZ lab by Dec 13, 2020.
> This is the installer branch we use for the installation.

But I don't understand what the PR is for? Do you want to merge this branch into git master for the installer?

**Comment by eranco74** [comment]
No longer relevant

**Pull request closed without merging**